primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -99,7 +103,12 @@ else if (submittedValue != null) {
 
             ValueHolder valueHolder = (ValueHolder) component;
             if (value == UNDEFINED_VALUE) {
-                value = valueHolder.getValue();
+                if (valueHolder instanceof HtmlOutputFormat) {
+                    value = encodeHtml((HtmlOutputFormat) valueHolder, context);
```

Comment:
You don't need to pass `(HtmlOutputFormat)valueHolder` here just pass `component` since its already a UIComponent that the signature of encodeHtml needs.